GeneratorInterface/Pythia8Interface/plugins/JetMatchingEWKFxFx.cc

```diff
@@ -0,0 +1,998 @@
+  }
+
+  if (!hjSlowJet->setup(tempEventJet)) {
+    errorMsg(
+        "Warning in JetMatchingMadgraph:matchPartonsToJets"
+        "Heavy: the SlowJet algorithm failed on setup");
+    return NONE;
+  }
+
+  while (hjSlowJet->sizeAll() - hjSlowJet->sizeJet() > 0) {
+    if (hjSlowJet->dNext() > qCutSq)
+      break;
+    hjSlowJet->doStep();
+  }
+
+  int nCLjets(0);
+  // Count the number of clusters with pT>qCut.  This includes the
+  //  original hard partons plus any hard emissions.
+  for (int idx = 0; idx < hjSlowJet->sizeAll(); ++idx) {
+    if (hjSlowJet->pT(idx) > sqrt(qCutSq))
+      nCLjets++;
+  }
+
+  // Debug printout.
+  if (MATCHINGDEBUG)
+    hjSlowJet->list(true);
+
+  // Count of the number of hadronic jets in SlowJet accounting
+  //  int nCLjets = nClus - nJets;
+  // Get number of partons. Different for MLM and FxFx schemes.
+  int nRequested = nParton;
+
+  // Veto event if too few hadronic jets
+  if (nCLjets < nRequested) {
+    if (MATCHINGDEBUG)
+      cout << "veto : hvy  LESS_JETS " << endl;
+    if (MATCHINGDEBUG)
+      cout << "nCLjets = " << nCLjets << "; nRequest = " << nRequested << endl;
+    return LESS_JETS;
+  }
+
+  // In exclusive mode, do not allow more hadronic jets than partons
+  if (exclusive) {
+    if (nCLjets > nRequested) {
+      if (MATCHINGDEBUG)
+        cout << "veto : excl hvy  MORE_JETS " << endl;
+      return MORE_JETS;
+    }
+  }
+
+  // No extra jets were present so no veto
+  return NONE;
+}
+
+int JetMatchingEWKFxFx::matchPartonsToJetsOther() {
+  // Currently, heavy jets are unmatched
+  // If there are no extra jets, then accept
+  // jetMomenta is NEVER used by MadGraph and is always empty.
+  //  This check does nothing.
+  //  Rather, if there is any heavy flavor that is harder than
+  //  what is present at the LHE level, then the event should
+  //  be vetoed.
+
+  // if (jetMomenta.empty()) return NONE;
+  // Count the number of hard partons
+  int nParton = typeIdx[2].size();
+
+  Event tempEventJet(workEventJet);
+
+  double scaleF(1.0);
+  // Rescale the heavy partons that are from the hard process to
+  //  have pT=collider energy.   Soft/collinear gluons will cluster
+  //  onto them, leaving a remnant of hard emissions.
+  for (int i = 0; i < nParton; ++i) {
+    scaleF = eventProcessOrig[0].e() / workEventJet[typeIdx[2][i]].pT();
+    tempEventJet[typeIdx[2][i]].rescale5(scaleF);
+  }
+
+  if (!hjSlowJet->setup(tempEventJet)) {
+    errorMsg(
+        "Warning in JetMatchingMadgraph:matchPartonsToJets"
+        "Heavy: the SlowJet algorithm failed on setup");
+    return NONE;
+  }
+
+  while (hjSlowJet->sizeAll() - hjSlowJet->sizeJet() > 0) {
+    if (hjSlowJet->dNext() > qCutSq)
+      break;
+    hjSlowJet->doStep();
+  }
+
+  int nCLjets(0);
+  // Count the number of clusters with pT>qCut.  This includes the
+  //  original hard partons plus any hard emissions.
+  for (int idx = 0; idx < hjSlowJet->sizeAll(); ++idx) {
+    if (hjSlowJet->pT(idx) > sqrt(qCutSq))
```

Comment:
```suggestion
    if (hjSlowJet->pT(idx) > qCut)
```
(or is it possible that `qCut` assumes negative values?)

Reply:
In principle qCut should not take negative values so indeed this does can be done.